detox/src/realms/DetoxPrimaryContext.js

```diff
@@ -210,12 +212,17 @@ class DetoxPrimaryContext extends DetoxContext {
 
     try {
       this[_lifecycleLogger].trace.end({ abortSignal: signal });
+      this[symbols.logger].close().catch(this[_logFinalError]);
       this[$logFinalizer].finalizeSync();
     } catch (err) {
-      this[symbols.logger].error({ err }, 'Encountered an error while merging the process logs:');
+      this[_logFinalError](err);
     }
   };
 
+  [_logFinalError] = (err) => {
+    this[_lifecycleLogger].error(err, 'Encountered an error while merging the process logs:');
+  };
```

Comment:
I don't fully understand what's happening here - need some help... It looks like a ["hail mary"](https://www.dictionary.com/browse/hail-mary), and if so, would it be helpful at all? Assuming the streams finalization didn't end well - what's going to happen when trying to write a new log? 🤔

Reply:
See where - in the console? I wonder whether that's true, seeing that all/some streams have been closed...

Reply:
Yes, in console. Only file streams are closed.